src/components/DocumentCard/DocumentCard.tsx

```diff
@@ -31,16 +27,30 @@ export class DocumentCard extends React.Component<IDocumentCardProps, any> {
           css(
             'ms-DocumentCard',
             {
-              'ms-DocumentCard--actionable': onClick ? true : false,
+              'ms-DocumentCard--actionable': actionable,
               'ms-DocumentCard--compact': type === DocumentCardType.compact ? true : false
             },
             className
           )
         }
-        onClick={ onClick }
+        onClick={ actionable ? this._onClick : null }
         style={ style }>
         { children }
       </div>
     );
   }
+
+  @autobind
+  private _onClick(ev: React.MouseEvent): void {
+    let { onClick, onClickHref } = this.props;
+
+    if (onClick) {
+      onClick(ev);
+    } else if (!onClick && onClickHref) {
```

Comment:
!onClick is unnecessary in this logic... it will always be false due to the "else"